build-system/tasks/visual-diff.js

```diff
@@ -0,0 +1,164 @@
+const gulp = require('gulp-help')(require('gulp'));
+const util = require('gulp-util');
+
+const percyCommand = 'percy snapshot';
+const defaultWidths = [375, 411];  // CSS widths: iPhone: 375, Pixel: 411.
+
+/**
+ * Executes the provided command; terminates this program in case of failure.
+ * Copied from pr-check.js.
+ * TODO(rsimha-amp): Refactor this into a shared library. Issue #9038.
+ *
+ * @param {string} cmd
+ */
+function execOrDie(cmd) {
+  const p =
+      child_process.spawnSync('/bin/sh', ['-c', cmd], {'stdio': 'inherit'});
+  if (p.status != 0) {
+    console/*OK*/.log(
+        `\n${fileLogPrefix}exiting due to failing command: ${cmd}`);
+    process.exit(p.status)
+  }
+}
+
+/**
+ * Extracts Percy project keys from the environment.
+ *
+ * @return {!Object} Object containing Percy project and token.
+ */
+function extractPercyKeys() {
+  // Repo slug to which to upload snapshots. Same as the Github repo slug.
+  let percyProject = '';
+  if (process.env.PERCY_PROJECT) {
+    percyProject = process.env.PERCY_PROJECT;
+  } else {
+    util.log(util.colors.red(
+        'PERCY_PROJECT must be specified as an environment variable'));
+    done();
+  }
+  util.log('Percy project: ', util.colors.magenta(percyProject));
+
+  // Secret token for the percy project.
+  let percyToken = '';
+  if (process.env.PERCY_TOKEN) {
+    percyToken = process.env.PERCY_TOKEN;
+  } else {
+    util.log(util.colors.red(
+        'PERCY_TOKEN must be specified as an environment variable'));
+    done();
+  }
+  util.log('Percy token: ', util.colors.magenta(percyToken));
+  return {
+    percyProject: percyProject,
+    percyToken: percyToken
+  };
+}
+
+/**
+ * Extracts Percy args from the command line.
+ *
+ * @return {!Object} Object containing extracted args.
+ */
+function extractPercyArgs() {
+  // Webpage to snapshot. This is a path, relative to amphtml/.
+  let webpage = '';
+  if (argv.webpage) {
+    webpage = argv.webpage;
+  } else {
+    console./*OK*/error(util.colors.red(
+        'Must specify a webpage to diff via --webpage'));
+    process.exit(1);
+  }
+  util.log('Webpage: ', util.colors.magenta(webpage));
+
+  // Smartphone screen widths to snapshot.
+  let widths = defaultWidths;
+  if (argv.widths) {
+    widths = JSON.parse("[" + argv.widths + "]");
+  }
+  util.log('Widths: ', util.colors.magenta(widths.toString()));
+
+  // TODO(rsimha): Separate out some test pages into directories, and then
+  // add an arg to include the directory containing assets for those pages.
+
+  return {
+    webpage: webpage,
+    widths: widths
+  };
+}
+
+/**
+ * Constructs the Percy command line with various args.
+ *
+ * @param {!Object} percyKeys Object containing access keys for the Percy repo.
+ * @return {String} Full command line to be executed.
+ */
+function constructCommandLine(percyKeys) {
+  let commandLine = [];
+
+  // Percy project keys.
+  commandLine.push('PERCY_PROJECT=' + percyKeys.percyProject);
```

Comment:
if we've already asserted that these are global env vars why do we need to export it again?